PeerioTechnologies/peerio-desktop
# chore: more conversion to typescript

#### Relevant info and issue/PR links

https://app.clubhouse.io/peerio/story/17336/desktop-complete-initial-conversion-to-typescript

some files were also refactored.

some broken or incorrect behaviour was documented or otherwise replicated as-is since we might be relying on it now; check `// FIXME` comments for some notes.

since most files are in TypeScript now, i've swapped the eslint configs so that TS files will be linted in-editor (but js files won't be). looks like the issues preventing simultaneous linting are almost fixed, but i tried upgrading `typescript-eslint-parser` and it wasn't too happy.

#### Testing instructions

most things will need to be tested; sections that were refactored could probably use a bit of extra attention. these are:

- functionality for dropping files into peerio to upload
- the "members" section in the sidebar
- the user picker (ie. the one used when adding new participants to a chat) **note that the layout on this will be broken pending https://github.com/PeerioTechnologies/peerio-desktop/pull/501/files being merged into this branch**
- the settings page tabs

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
